agent-inject/agent/annotations_test.go

```diff
@@ -135,20 +135,60 @@ func TestSecretAnnotations(t *testing.T) {
 		if tt.expectedKey != "" {
 			if len(agent.Secrets) == 0 {
 				t.Error("Secrets length was zero, it shouldn't have been")
+			}
+			if agent.Secrets[0].Name != tt.expectedKey {
+				t.Errorf("expected %s, got %s", tt.expectedKey, agent.Secrets[0].Name)
+			}
 
-				if agent.Secrets[0].Name != tt.expectedKey {
-					t.Errorf("expected %s, got %s", tt.expectedKey, agent.Secrets[0].Name)
-				}
-
-				if agent.Secrets[0].Path != tt.expectedPath {
-					t.Errorf("expected %s, got %s", tt.expectedPath, agent.Secrets[0].Path)
+			if agent.Secrets[0].Path != tt.expectedPath {
+				t.Errorf("expected %s, got %s", tt.expectedPath, agent.Secrets[0].Path)
 
-				}
 			}
 		} else if len(agent.Secrets) > 0 {
 			t.Error("Secrets length was greater than zero, it shouldn't have been")
 		}
+	}
+}
+
+func TestSecretAnnotationsWithPreserveCaseSensitivityFlagOn(t *testing.T) {
+	tests := []struct {
```

Comment:
@jasonodonnell this is not dry but making it dry would make tests a bit complicated so I duplicated the code